common/utils.go

```diff
@@ -115,29 +113,60 @@ func hashWithSalt(password string, salt []byte, iterations int, algorithm string
 	buffer.WriteString(fmt.Sprintf("%s$", algorithm))
 	buffer.WriteString(strconv.Itoa(iterations))
 	buffer.WriteString("$")
-	buffer.WriteString(base64.StdEncoding.EncodeToString(salt))
+	// Re-encode salt, using encoding supplied in saltEncoding param
+	if saltEncoding == "utf-8" {
+		buffer.WriteString(string(salt))
+
+	} else {
+		buffer.WriteString(base64.StdEncoding.EncodeToString(salt))
+	}
```

Comment:
Here we could have a check for the different encodings, while exported valid constants (or a map) could be used to check at backends initialization too, e.g.:
```
// Declare the valid encodings for validation.
const (
    UTF8 = "utf-8"
    Base64 = "base64"
)

...

switch saltEncoding{
  case UTF8:
    ...
  case Base64:
    ...
  default:
    // Error out here.
}
```

Reply:
Modified approach used in most recent commit.